### ISSUE TITLE
E2E Tests (Firefox) Failing Consistently

### ISSUE THREAD

**Issue by samuelfangjw** [TEAMMATES/teammates#11591]
- **Environment**: 
`master` branch at commit [f70ae53](https://github.com/TEAMMATES/teammates/commit/f70ae535d0d0586d48530704227b67f169f49bfb).

The same 4 [E2E](https://github.com/TEAMMATES/teammates/actions/workflows/e2e.yml) tests have been failing consistently over the last 24 hours, including the [daily scheduled one](https://github.com/TEAMMATES/teammates/runs/5296728447?check_suite_focus=true).

**Additional info**
- ~Tests pass locally on my machine~
- From the stack trace, the issue could be related to this method `teammates.e2e.pageobjects.Browser.closeCurrentWindowAndSwitchToParentWindow(Browser.java:114)`. The 4 failing tests call this method. Tests which do not call this method have no issues.
- Stacktrace from one failure, the others are similar.

```
org.gradle.internal.serialize.PlaceholderException: Unable to establish websocket connection to http://localhost:42277/devtools/browser/3018d1f6-b00d-4a4c-84b9-73ad5e94abcc
Build info: version: '4.0.0', revision: '3a21814679'
System info: host: 'fv-az173-681', ip: '10.1.1.44', os.name: 'Linux', os.arch: 'amd64', os.version: '5.11.0-1028-azure', java.version: '11.0.14'
Driver info: driver.version: RemoteWebDriver
    org.openqa.selenium.remote.http.netty.NettyWebSocket.<init>(NettyWebSocket.java:104)
    org.openqa.selenium.remote.http.netty.NettyWebSocket.lambda$create$3(NettyWebSocket.java:137)
    org.openqa.selenium.remote.http.netty.NettyClient.openSocket(NettyClient.java:127)
    org.openqa.selenium.devtools.Connection.<init>(Connection.java:77)
    org.openqa.selenium.firefox.FirefoxDriver.maybeGetDevTools(FirefoxDriver.java:359)
    org.openqa.selenium.remote.RemoteWebDriver.close(RemoteWebDriver.java:441)
    teammates.e2e.pageobjects.Browser.closeCurrentWindowAndSwitchToParentWindow(Browser.java:114)
    teammates.e2e.pageobjects.AppPage.closeCurrentWindowAndSwitchToParentWindow(AppPage.java:608)
    teammates.e2e.cases.InstructorStudentListPageE2ETest.testAll(InstructorStudentListPageE2ETest.java:97)
    ...
```

Edit: This seems to be related to recent firefox 97 release breaking some features on selenium. Should be fixed by updating selenium to 4.1.3 when it is released.

Edit 2: #11611 is a temporary fix for this issue. The updated selenium dependency is expected in the next few days. Once it is out, we can revert the temporary fix, update the dependency and close this PR.

**Comment by fsgmhoward** [comment]
Is it possible to run the E2E test with older version of Firefox? This issue is quite urgent as we have a list of PRs failing this test currently. It is marked as a "required" test so its reliability should be essential.

**Comment by samuelfangjw** [comment]
> Is it possible to run the E2E test with older version of Firefox? This issue is quite urgent as we have a list of PRs failing this test currently. It is marked as a "required" test so its reliability should be essential.

I think this is a good idea, I've implemented this in #11611.
